### ISSUE TITLE
loss function

### ISSUE THREAD

**Issue by Biyhjj** [JinshuaiBai/LSWR_loss_function_PINN#2]
Dear Dr.Bai

I would like to say the idea behind your work is impressive!And i am trying to predict the U and V according to 01_pure_bend you have introduced.But I am confused about a few points.
1. in Dif_op_u and Dif_op_v , why one U = temp*x , one U = temp?
2. in Opt ,  is y in L2 also the output of the PINN module ? and why is it calculated this way?

Thanks for replying

**Comment by JinshuaiBai** [comment]
Thank you for your nice words. 
1. Dif_op_u and Dif_op_v are the functions to calculate the derivatives of displacement u and v, respectively. Recall the pure bending beam case, the left boundary needs to satisfy that u(x=0)=0. In this case, we apply the hard boundary condition technique in PINN; that is, we let the output of FNN as intermediate output and multiply it with x ( u=FNN(x,y)*x ). By doing so, the boundary condition u(x=0)=0 can be naturally satisfied without introducing additional loss terms to impose this condition. It is worth highlighting that the training process will be more challenging (harder to converge) if more loss terms are introduced. On the other hand, the displacement v doesn't need to satisfy any hard boundary condition. Therefore, the U=temp is directly applied without multiplying x or y.  
2.  The list y is not the output of the PINN, they are ground truths of the problem; that is, the traction boundary conditions of the pure bending beam. Those data are known before training, and they are used to correct the stress boundary condition predictions from the PINN. You can refer to basic elastic mechanics books for more details.  

Hope this can answer your questions.

**Comment by Biyhjj** [comment]
Thanks a million!
May the joy and happiness around you today and always!

**Comment by jackliuzefeng** [comment]
Dr.bai,请问为什么求损失对于L1面积权重为dx的平方，不应该按照论文中的三角形面积权重进行计算吗

**Comment by jackliuzefeng** [comment]
here is the code in pure_bend\Opt
![P}J9)O8`RR1U6Q(6R L3N Y](https://github.com/user-attachments/assets/ddc90764-ca00-4640-b8f9-9bdd66e67450)

**Comment by JinshuaiBai** [comment]
> Dr.bai,请问为什么求损失对于L1面积权重为dx的平方，不应该按照论文中的三角形面积权重进行计算吗

您好，三角形面积权重计算适用于不规则的粒子点分布情况的，对于规则几何，我还是更推荐使用高斯积分点做数值积分。由于pure_bend里面的我使用的是一阶的高斯积分点，因此，每个点的权重都是相等的，就是点所占面积

希望这个回答可以帮助到你

**Comment by jackliuzefeng** [comment]
非常感谢您的回复，对我帮助很大，祝您生活愉快，工作顺利，发论文如流水(^-^)👍



---Original---
From: "Jinshuai ***@***.***&gt;
Date: Fri, Nov 22, 2024 08:19 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [JinshuaiBai/LSWR_loss_function_PINN] loss function (Issue #2)




  
Dr.bai,请问为什么求损失对于L1面积权重为dx的平方，不应该按照论文中的三角形面积权重进行计算吗
  
您好，三角形面积权重计算适用于不规则的粒子点分布情况的，对于规则几何，我还是更推荐使用高斯积分点做数值积分。由于pure_bend里面的我使用的是一阶的高斯积分点，因此，每个点的权重都是相等的，就是点所占面积
 
希望这个回答可以帮助到你

**Comment by JinshuaiBai** [comment]
> 非常感谢您的回复，对我帮助很大，祝您生活愉快，工作顺利，发论文如流水(^-^)👍
> […](#)
> ---Original--- From: "Jinshuai ***@***.***&gt; Date: Fri, Nov 22, 2024 08:19 AM To: ***@***.***&gt;; Cc: ***@***.******@***.***&gt;; Subject: Re: [JinshuaiBai/LSWR_loss_function_PINN] loss function (Issue #2) Dr.bai,请问为什么求损失对于L1面积权重为dx的平方，不应该按照论文中的三角形面积权重进行计算吗 您好，三角形面积权重计算适用于不规则的粒子点分布情况的，对于规则几何，我还是更推荐使用高斯积分点做数值积分。由于pure_bend里面的我使用的是一阶的高斯积分点，因此，每个点的权重都是相等的，就是点所占面积 希望这个回答可以帮助到你 — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you commented.Message ID: ***@***.***&gt;

十分感谢您的支持和祝福，也祝您一切顺利！
